test/node/bson_binary_vector.spec.test.ts

```diff
@@ -0,0 +1,223 @@
+    Int8: 0x03,
+    Float32: 0x27,
+    PackedBit: 0x10
+  } as const);
+
+  if (vector.sub_type !== 9) return;
+
+  const size = vector.position;
+  const d_type = vector.buffer[0] ?? 0;
+  const padding = vector.buffer[1] ?? 0;
+
+  if ((d_type === VECTOR_TYPE.Float32 || d_type === VECTOR_TYPE.Int8) && padding !== 0) {
+    throw new BSONError('Invalid Vector: padding must be zero for int8 and float32 vectors');
+  }
+
+  if (d_type === VECTOR_TYPE.PackedBit && padding !== 0 && size === 2) {
+    throw new BSONError(
+      'Invalid Vector: padding must be zero for packed bit vectors that are empty'
+    );
+  }
+
+  if (d_type === VECTOR_TYPE.PackedBit && padding > 7) {
+    throw new BSONError(
+      `Invalid Vector: padding must be a value between 0 and 7. found: ${padding}`
+    );
+  }
+}
+
+function fixFloats(f: string | number): number {
+  if (typeof f === 'number') {
+    return f;
+  }
+  if (f === 'inf') {
+    return Infinity;
+  }
+  if (f === '-inf') {
+    return -Infinity;
+  }
+  throw new Error(`test format error: unknown float value: ${f}`);
+}
+
+function fixInt8s(f: number | string): number {
+  if (typeof f !== 'number') throw new Error('test format error: unexpected test data');
+
+  if (f < -128 || f > 127) {
+    // Javascript Int8Array only supports values from -128 to 127
+    throw new Error(`unsupported_error: int8 out of range: ${f}`);
+  }
+  return f;
+}
+
+function fixBits(f: number | string): number {
+  if (typeof f !== 'number') throw new Error('test format error: unexpected test data');
+
+  if (f > 255 || f < 0 || !Number.isSafeInteger(f)) {
+    // Javascript Uint8Array only supports values from 0 to 255
+    throw new Error(`unsupported_error: bit out of range: ${f}`);
+  }
+  return f;
+}
+
+function make(
+  vector: (number | string)[],
+  dtype_hex: string,
+  dtype_alias: DTypeAlias,
+  padding: number
+): Binary {
+  let binary: Binary;
+  switch (dtype_alias) {
+    case 'PACKED_BIT':
+    case 'INT8': {
+      const array = new Int8Array(vector.map(dtype_alias === 'INT8' ? fixInt8s : fixBits));
+      const buffer = new Uint8Array(array.byteLength + 2);
+      buffer[0] = +dtype_hex;
+      buffer[1] = padding;
+      buffer.set(new Uint8Array(array.buffer), 2);
+      binary = new Binary(buffer, 9);
+      break;
+    }
+
+    case 'FLOAT32': {
+      const array = new Float32Array(vector.map(fixFloats));
+      const buffer = new Uint8Array(array.byteLength + 2);
+      buffer[0] = +dtype_hex;
+      buffer[1] = padding;
+      if (isBigEndian) {
+        for (let i = 0; i < array.length; i++) {
+          const bytes = new Uint8Array(array.buffer, i * 4, 4);
+          bytes.reverse();
+          buffer.set(bytes, i * 4 + 2);
+        }
+      } else {
+        buffer.set(new Uint8Array(array.buffer), 2);
+      }
+      binary = new Binary(buffer, 9);
+      break;
+    }
+
+    default:
+      throw new Error(`Unknown dtype_alias: ${dtype_alias}`);
+  }
+
+  binary.buffer[0] = +dtype_hex;
```

Comment:
this is redundant - all cases in the switch already do this.

Reply:
Fair, removed the same lines from inside the switch